src/Payum/Core/CoreGatewayFactory.php

```diff
@@ -59,10 +63,26 @@ public function createConfig(array $config = array())
         $config['twig.env'] = null;
 
         $config->defaults(array(
-            'payum.template.layout' => '@PayumCore/layout.html.twig',
+            'httplug.client'=>function(ArrayObject $config) {
+                return HttpClientDiscovery::find();
+            },
+            'httplug.message_factory'=>function(ArrayObject $config) {
+                return MessageFactoryDiscovery::find();
+            },
+            'payum.http_client'=>function(ArrayObject $config) {
+                $func = $config['httplug.client'];
+                if (is_callable($func)) {
+                    $func = $func($config);
+                }
```

Comment:
The test has to be fixed in that case. You should do a call_user_func there in the test.